dfm/cleaning/deduper.py

```diff
@@ -240,7 +243,7 @@ def deduplicate(
 
         # Convert corpus to an iterable of strings if a Dataset is given
         if isinstance(corpus, Dataset) or isinstance(corpus, IterableDataset):
-            corpus = [sample["text"] for sample in corpus]
+            corpus = (sample["text"] for sample in corpus)
```

Comment:
I wondered why this changed originally, should have asked then. I will make sure to do so next time